--- a/Lib/distutils/tests/test_build_ext.py
+++ b/Lib/distutils/tests/test_build_ext.py
@@ -66,8 +66,11 @@
             if runshared is None:
                 cmd.library_dirs = ['.']
             else:
-                name, equals, value = runshared.partition('=')
-                cmd.library_dirs = value.split(os.pathsep)
+                if sys.platform == 'darwin':
+                    cmd.library_dirs = []
+                else:
+                    name, equals, value = runshared.partition('=')
+                    cmd.library_dirs = value.split(os.pathsep)
 
     @unittest.skipIf(not os.path.exists(_XX_MODULE_PATH),
                      'xxmodule.c not found')
